Makefile

```diff
@@ -1605,19 +1605,23 @@ llama-q8dot: pocs/vdot/q8dot.cpp ggml/src/ggml.o \
 # Mark legacy binary targets as .PHONY so that they are always checked.
 .PHONY: main quantize perplexity embedding server
 
+# Define the object file target
+examples/deprecation-warning/deprecation-warning.o: examples/deprecation-warning/deprecation-warning.cpp
+	$(CXX) $(CXXFLAGS) -c $< -o $@
+
 # NOTE: We currently will always build the deprecation-warning `main` and `server` binaries to help users migrate.
 #  Eventually we will want to remove these target from building all the time.
-main: examples/deprecation-warning/deprecation-warning.cpp
+main: examples/deprecation-warning/deprecation-warning.o
 	$(CXX) $(CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)
```

Comment:
I think you should remove these `$(CXX) $(CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)` in all legacy targets, as they will use the examples/deprecation-warning/deprecation-warning.o object, so there is no need to compile and overwrite it again